### PR TITLE
Add Java publish workflow

### PR DESCRIPTION
**Pull request by samuel-rufi**

### CODE REVIEW

File: .github/workflows/java_binding_publish.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
+name: Publish Java binding
+
+on:
+  push:
+    branches:
+      - staging
```
Comment (Thoralf-M) [inline .github/workflows/java_binding_publish.yml]: :carloshuh:

Comment (samuel-rufi) [reply .github/workflows/java_binding_publish.yml]: what should i replace it with? :carlos:

Comment (Thoralf-M) [reply .github/workflows/java_binding_publish.yml]: develop? We have no staging branch

Comment (Thoralf-M) [reply .github/workflows/java_binding_publish.yml]: But should probably also not run on every push

Comment (Thoralf-M) [reply .github/workflows/java_binding_publish.yml]: Ok maybe we will get staging :trollface:

**Pull request closed without merging**